mantidproject/mantid
# Upgrade GSL to v2 on Windows

**Description of work.**

Updates GSL library to v2.5 on Windows bundle.

The system tests had various checks regarding the GSL version that were all based on the OS rather than what was configured at build time. This has been fixed by creating a Python module in the `mantid` package that contains information about the mantid build configuration and allows a user to check the version of GSL directly rather than assuming an OS has a certain version. 

**To test:**

All system tests should pass. Ideally run the `ISISIndirectInelastic.IRISConvFit` system test on Windows.

Fixes #26262 

*This does not require release notes* because **this is an internal change.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
